server/client.go

```diff
@@ -3192,10 +3225,40 @@ func (c *client) Account() *Account {
 // prunePerAccountCache will prune off a random number of cache entries.
 func (c *client) prunePerAccountCache() {
 	n := 0
-	for cacheKey := range c.in.pacache {
-		delete(c.in.pacache, cacheKey)
-		if n++; n > prunePerAccountCacheSize {
-			break
+	if time.Now().UnixNano()-c.in.losc >= int64(orphanSubsCheckInterval) {
+		for cacheKey, pac := range c.in.pacache {
+			canRemove := true
```

Comment:
If we have any sub or qsub as part of a result that has been closed we can invalidate the result and delete the whole thing. It will come back properly if used again.

Reply:
True.. it is a cache and ok to remove even when valid. I will change that.